### PR TITLE
[WIP] push slice to prismic spinners

### PR DESCRIPTION
**Pull request by MarcMcIntosh**
Found some issues when converting old code to type script, the spinners them self are easy to sort out. the types are more difficult

### CODE REVIEW

**Comment by MarcMcIntosh** [comment]
Note: ideally both loadiing and image-loading wouldn't be set at the same time, but this require further refactoring off `packages/slice-machine/lib/models/ui/Store.ts` and everything that uses it. I've added refactoring this to the technical debt board

File: packages/slice-machine/lib/builders/SliceBuilder/Header/index.tsx [hunk 1]
```diff
@@ -55,91 +70,62 @@ const Header = ({ Model, store, variation, onSave, onPush, isLoading }) => {
...
-                    {
-                      // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
-                      Model.infos.sliceName
-                    }{" "}
-                    /{" "}
-                    {
-                      // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
-                      variation.name
-                    }
+                    {`/ ${Model.infos.sliceName} / ${variation.name}`}
```
Comment (Heyrows) [inline packages/slice-machine/lib/builders/SliceBuilder/Header/index.tsx:74]: why do you need string interpolation ? wasn't it good before ?

File: packages/slice-machine/lib/builders/SliceBuilder/Header/index.tsx [hunk 2]
```diff
@@ -55,91 +70,62 @@ const Header = ({ Model, store, variation, onSave, onPush, isLoading }) => {
...
           />
           <VariationModal
             isOpen={showVariationModal}
             onClose={() => setShowVariationModal(false)}
             onSubmit={(id, name, copiedVariation) => {
-              // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-unsafe-call
               store.copyVariation(id, name, copiedVariation);
-              // eslint-disable-next-line @typescript-eslint/no-floating-promises
-              router.push(
+              return router.push(
```
Comment (Heyrows) [inline packages/slice-machine/lib/builders/SliceBuilder/Header/index.tsx:122]: Note: if you don't want to return it, I think you can do 
```suggestion
              void router.push(
```

File: packages/slice-machine/lib/builders/SliceBuilder/SideBar/components/FooterButton.tsx [hunk 3]
```diff
@@ -1,9 +1,15 @@
-import { Button, Flex, Spinner } from "theme-ui";
```
Comment (Heyrows) [inline packages/slice-machine/lib/builders/SliceBuilder/SideBar/components/FooterButton.tsx]: Remove the file if it's dead code, no need to transform it to typescript

File: packages/slice-machine/lib/builders/SliceBuilder/Header/SaveButton.tsx [hunk 4]
```diff
@@ -0,0 +1,52 @@
+import React from "react";
+import { Spinner, Button } from "theme-ui";
+
+const SaveButton: React.FC<{
+  __status: string;
+  isTouched?: boolean;
+  onSave: () => void;
+  onPush: () => void;
+  isLoading: boolean;
+  imageLoading: boolean;
```
Comment (Heyrows) [inline packages/slice-machine/lib/builders/SliceBuilder/Header/SaveButton.tsx]: call it disabled please, it's a button and it's used that way.

File: packages/slice-machine/lib/builders/SliceBuilder/index.tsx [hunk 5]
```diff
@@ -110,6 +110,7 @@ const SliceBuilder: React.FunctionComponent = () => {
         // @ts-expect-error
         onSave={() => store.save(Model, setData)}
         isLoading={data.loading}
+        imageLoading={data.imageLoading}
```
Comment (Heyrows) [inline packages/slice-machine/lib/builders/SliceBuilder/index.tsx:113]: I understand why you can't change the shape of data, that would hard.
But please change Header parameter's name, we don't want to pass `imageLoading` everywhere.

**Review by Heyrows — CHANGES_REQUESTED** [review]

File: packages/slice-machine/lib/builders/SliceBuilder/Header/index.tsx (hunk 1 above)
Comment (MarcMcIntosh) [reply packages/slice-machine/lib/builders/SliceBuilder/Header/index.tsx:74]: same thing, just easier to read

File: packages/slice-machine/lib/builders/SliceBuilder/Header/index.tsx (hunk 2 above)
Comment (MarcMcIntosh) [reply packages/slice-machine/lib/builders/SliceBuilder/Header/index.tsx:122]: I think return works better as the process is async

File: packages/slice-machine/lib/builders/SliceBuilder/Header/index.tsx (hunk 1 above)
Comment (Heyrows) [reply packages/slice-machine/lib/builders/SliceBuilder/Header/index.tsx:74]: ok 👍 , did you know that it's prettier that adds the `{" "}` in the tsx XD ?

File: packages/slice-machine/lib/builders/SliceBuilder/Header/index.tsx (hunk 2 above)
Comment (Heyrows) [reply packages/slice-machine/lib/builders/SliceBuilder/Header/index.tsx:122]: Yeah but we don't need what the function returns.
I'm supposing you put `return` for eslint.
The good way to do it for a promise on which you don't need the result is void.

But that's a `Note`, so don't change it if you don't want to. That was just me trying to spread what i consider best practices 😃

File: packages/slice-machine/lib/builders/SliceBuilder/Header/index.tsx (hunk 1 above)
Comment (MarcMcIntosh) [reply packages/slice-machine/lib/builders/SliceBuilder/Header/index.tsx:74]: nope, I didn't know that

**Review by Heyrows — APPROVED** [review]
I didn't require so much extraction out of the button, just the rename of the disable.
But it's fine and it does looks nicer than before.

**Pull request merged**